source/simulator/core.cc

```diff
@@ -849,17 +851,46 @@ namespace aspect
         {
           Assert (is_element (*p, geometry_model->get_used_boundary_indicators()),
                   ExcInternalError());
-          VectorTools::interpolate_boundary_values (dof_handler,
-                                                    *p,
-                                                    VectorFunctionFromScalarFunctionObject<dim>(std_cxx11::bind (&BoundaryTemperature::Interface<dim>::temperature,
-                                                        std_cxx11::cref(*boundary_temperature),
-                                                        std_cxx11::cref(*geometry_model),
+
+          if (std::isnan(boundary_temperature->boundary_temperature(*p,geometry_model->representative_point(0.0))))
+            {
+              if (timestep_number == 0)
+                pcout << "WARNING: Your selected boundary temperature plugin uses the deprecated "
+                      << "interface function 'temperature(geometry_model,boundary_indicator,position)'. "
+                      << "This function will be removed in a future ASPECT "
+                      << "release. Please update your plugin NOW to the new interface "
+                      << "'boundary_temperature(boundary_indicator,position) "
+                      << "instead, and derive it from SimulatorAccess, if you need access to the "
+                      << "geometry model." << std::endl;
```

Comment:
I'd just abort here. But in fact, you could already do that if you happen to end up in that function as mentioned in my comment there.